content/blog/posts/vectors-matrices-matrix-order.md

```diff
@@ -0,0 +1,732 @@
+---
+title: "Vectors, Matrices & Matrix Order"
+date: 2023-02-22
+slug: vectors-matrices-matrix-order
+author: John Coimbra Walsh
+blog_img: "/images/blog/dev-series.JPG"
+full_img: ""
+---
+
+In this post I will attempt to demystify a topic that is the source of confusion for a lot of people as they dip their toes into learning the concepts that underpin computer graphics: vectors, matrices and matrix order.
+
+## Matrices & vectors
+
+An \\(n \cdot m\\) matrix consists of \\(n\\) rows and \\(m\\) columns. The product of two matrices \\(\mathbf{A}\\) and \\(\mathbf{B}\\) is well defined when the number of columns in \\(\mathbf{A}\\) matches the number of rows in \\(\mathbf{B}\\). Their product is a matrix with the number of rows of \\(\mathbf{A}\\) and the number of columns of \\(\mathbf{B}\\). Thus, for a given \\(2×3\\) matrix \\(\mathbf{A}\\) to be multiplied with a given \\(3×2\\) matrix \\(\mathbf{B}\\), the product is calculated as follows to produce the resulting \\(2×2\\) matrix:
+
+$$
+\begin{array}{l@{}l}
+\mathbf{A} \cdot \mathbf{B} 
+    &{}= \begin{bmatrix} a_{11} & a_{12} & a_{13} \\\ a_{21} & a_{22} & a_{23} \end{bmatrix} \cdot \begin{bmatrix} b_{11} & b_{12} \\\ b_{13} & b_{21} \\\ b_{22} & b_{23} \end{bmatrix} \\\ \\\
+    &{}= \begin{bmatrix} a_{11}b_{11} + a_{12}b_{21} + a_{13}b_{31} & a_{11}b_{12} + a_{12}b_{22} + a_{13}b_{32} \\\ 
+a_{21}b_{11} + a_{22}b_{21} + a_{23}b_{31} & a_{21}b_{12} + a_{22}b_{22} + a_{23}b_{32} \end{bmatrix}
+\end{array}
+$$
+
+Vectors can be written as either rows:
+
+$$\vec{r} = \begin{bmatrix} r_{1} & r_{2} & r_{3} & ... & r_{m}  \end{bmatrix}$$
+
+Or columns:
+
+$$\vec{c} = \begin{bmatrix} c_{1} \\\ c_{2} \\\ c_{3} \\\ ... \\\ c_{m}  \end{bmatrix}$$
+
+The vector-matrix product follows the same rules as the matrix product, albeit treating row vectors as \\(1×m\\) matrices and column vectors as \\(m×1\\) matrices. Thus, for a given \\(2×2\\) matrix \\(\mathbf{M}\\) to be multiplied with a given \\(1×2\\) row vector \\(\vec{r}\\), we must ensure that the number of columns of the matrix on the left hand side matches the number of rows of the matrix on the right hand side. We achieve this by pre-multiplying the vector \\(\vec{r}\\) with \\(\mathbf{M}\\), resulting in a \\(1×2\\) matrix:
+
+$$
+\begin{array}{l@{}l}
+\vec{r} \cdot \mathbf{M} 
+    &{}= \begin{bmatrix} r_{1} & r_{2} \end{bmatrix} \cdot \begin{bmatrix} m_{11} & m_{12} \\\ m_{21} & m_{22}  \end{bmatrix} \\\ \\\
+    &{}= \begin{bmatrix} 
+r_{1}m_{11} + r_{2}m_{21} & 
+r_{1}m_{12} + r_{2}m_{22} \end{bmatrix}
+\end{array}
+$$
+
+In order to multiply a column vector by the same matrix, we must ensure that the number of columns of the matrix on the left hand side matches the number of rows of the matrix on the right hand side. We achieve this by post-multiplying the vector \\(\vec{c}\\) with \\(\mathbf{M}\\), resulting in a \\(2×1\\) matrix:
```

Comment:
I'd say it's a little confusing to call it the same matrix. It is the same matrix, but it will be a different transformation when you work with a column vector vs a row vector.

Reply:
```suggestion
In order to multiply a column vector, we must ensure that the number of columns of the matrix on the left hand side matches the number of rows of the matrix on the right hand side. We achieve this by post-multiplying the vector \\(\vec{c}\\) with \\(\mathbf{M}\\), resulting in a \\(2×1\\) matrix:
```